### ISSUE TITLE
Add elevator

### ISSUE THREAD

**Issue by aidnem** [team401/2025-Robot-Code#4]
**Purpose**
The purpose of this addition is to add an elevator subsystem and a mechanism and IO for the elevator.

**Project Scope**
- [x] Add an elevator subsystem (we might rename this later when we add in the arm, since the arm and elevator will probably be mechanisms in the same subsystem).
- [x] Implement an elevator mechanism that lets elevator be commanded to a position, and can return the elevator position to anything that needs it.
- [x] Add a sim elevator IO that simulates a physical elevator and the CRT encoder values.
- [x] Make elevator mechanism convert CRT encoder values.
Later:
- [ ] Add hardware IO that behaves the same as sim

**Comment by PChild** [comment]
Can we try Motion Magic Expo for the elevator please? https://v6.docs.ctr-electronics.com/en/stable/docs/api-reference/device-specific/talonfx/motion-magic.html#motion-magic-expo

**Comment by aidnem** [comment]
Sure!

**Comment by PChild** [comment]
@jkleiber is someone assigned to do sim for this?

**Comment by jkleiber** [comment]
@PChild I believe @aidnem is working on the sim for it

**Comment by aidnem** [comment]
Yes, I'm working on sim. There's a basic version implemented right now but I think the gear math is wrong because some of the numbers don't make sense.
